### PR TITLE
refactoring to monorepo

### PR DESCRIPTION
**Pull request by urish**
Using [yarn workspaces](https://yarnpkg.com/lang/en/docs/workspaces/) and [lerna](https://github.com/lerna/lerna), similar to what [jest](https://github.com/facebook/jest/) and [babel](https://github.com/babel/babel) do.

Also renaming typewiz to typewiz-core (see #54). I'd love to get some community feedback before this is merged - perhaps @styfle, @Bnaya can review my setup?

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,50 +1,38 @@
 {
-  "name": "typewiz",
```
Comment (Bnaya) [inline package.json]: it's recommended to add engines entry with minimal yarn version. yarn do validate it
When workspaces involved, i would go for at least `1.3.2`

File: packages/typewiz-core/package.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  "version": "0.7.2",
+  "main": "dist/index.js",
+  "typings": "dist/index.d.ts",
+  "repository": "https://github.com/urish/typewiz",
+  "author": "Uri Shaked <uri@urishaked.com>",
+  "license": "MIT",
+  "scripts": {
+    "build": "rimraf dist && tsc && copyfiles -u 1 src/type-collector-snippet.ts dist",
+    "lint": "tslint -p .",
+    "prepublish": "npm run build"
```
Comment (Bnaya) [inline packages/typewiz-core/package.json]: npm -> yarn

File: package.json (hunk 1 above)
Comment (urish) [reply package.json]: good point, I will add

File: packages/typewiz-core/package.json (hunk 2 above)
Comment (urish) [reply packages/typewiz-core/package.json]: good catch!

**Comment by urish** [comment]
Thank you so much @Bnaya! I plan to publish through my local machine, any pointers on how to do that?

If I understand correctly, running `lerna publish --conventional-commits` should do the trick - bump versions, create a release tag, publish to NPM and create / update the changelog?

**Pull request merged**